### PR TITLE
Refactor imports to explicitly overload methods

### PR DESCRIPTION
**Pull request by odow**
There were lots of imports scattered around the code that make it hard to know what functions were defined by Convex, and what were overloaded. One example is `Base.sign`, which was imported in `src/expressions.jl`. 

The exports, e.g., from LinearAlgebra, are retained.<s>, so the only breaking part about this is `Convex.sign` to `Base.sign`, and we can't fix this by exporting because it's a `Base` method.</s>

### CODE REVIEW

File: src/Convex.jl [hunk 1]
```diff
@@ -1,62 +1,115 @@
...
+    :norm2,
+    :tr,
+)
+    @eval begin
+        using LinearAlgebra: $k
+        export $k
+    end
+end
+using AbstractTrees: children
+using Base: sign
```
Comment (blegat) [inline src/Convex.jl:112]: Why is this line needed ? You explicitly do `Base.sign`.

Comment (odow) [reply src/Convex.jl:112]: Because docs and tests assume that there is `Convex.sign`. Same for `Convex.children`. I didn't add a comment because this is tested.

Comment (blegat) [reply src/Convex.jl:112]: Ah ok, so we need it for backward compatibility

Comment (odow) [reply src/Convex.jl:112]: Yeah. Before tagging the next release, there are a bunch of deprecation things we could remove. This would be one candidate.

Comment (odow) [reply src/Convex.jl:112]: But that's for a different PR.
